File: DotNet/Framework/AutomationFramework/Helpers/Drivers/Edge.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Edge;

namespace Helpers.Drivers
{
    public class Edge : IDriver
    {
        IWebDriver driver;

        public Edge()
        {
        }
        public void CloseDriver()
        {
            driver.Close();
            driver.Quit();
        }

        public IWebDriver GetDriver()
        {
            return driver;
        }

        public void MaximizeDriver()
        {
            driver.Manage().Window.Maximize();
        }

        public void MinimizeDriver()
        {
            driver.Manage().Window.Minimize();
        }

        public void NavigateURL(string url)
        {
            driver.Navigate().GoToUrl(url);
        }

        public void SetDriver()
        {
            this.driver = new EdgeDriver();
        }
    }
}

File: Java/AutomationFramework/src/com/app/helpers/drivers/Edge.java
package com.app.helpers.drivers;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.edge.EdgeDriver;

public class Edge implements IDriver{
    WebDriver driver;

    @Override
    public WebDriver GetDriver() {
        if(driver == null)
            SetDriver();

        return driver;
    }

    @Override
    public void SetDriver() {
        String path = System.getProperty("user.dir");
        System.setProperty("webdriver.edge.driver", path + "\\drivers\\msedgedriver.exe");

        this.driver = new EdgeDriver();
    }

    @Override
    public void MaximizeDriver() {
        driver.manage().window().maximize();
    }

    @Override
    public void MinimizeDriver() {
        driver.manage().window().minimize();
    }

    @Override
    public void NavigateURL(String url) {
        driver.get(url);
    }

    @Override
    public void CloseDriver() { driver.close(); }

    @Override
    public void QuitDriver() { driver.quit(); }
}
